agent/consul/state/intention.go

```diff
@@ -207,6 +207,294 @@ func (s *Store) legacyIntentionsListTxn(tx ReadTxn, ws memdb.WatchSet, entMeta *
 
 var ErrLegacyIntentionsAreDisabled = errors.New("Legacy intention modifications are disabled after the config entry migration.")
 
+func (s *Store) IntentionMutation(idx uint64, op structs.IntentionOp, mut *structs.IntentionMutation) error {
+	tx := s.db.WriteTxn(idx)
+	defer tx.Abort()
+
+	usingConfigEntries, err := areIntentionsInConfigEntries(tx, nil)
+	if err != nil {
+		return err
+	}
+	if !usingConfigEntries {
```

Comment:
I'm not following why this error is being returned. Isn't `!usingConfigEntries` true before the migration? `ErrLegacyIntentionsAreDisabled` is about legacy intentions being disabled after the migration.

It seems like this error should be about not being able to mutate config intentions before intentions are migrated.

Reply:
Short answer is belt-and-suspenders checks upon checks.

None of the RPC endpoints will let you get far enough to write something into the raft log that would trigger `Store.IntentionMutation` until after the migration has happened.

This is just insurance so that even if we missed a spot it still wouldn't work.

Reply:
Right, we should still have the check, I'm wondering if the error message should be different

Reply:
Changing it.